File: src/test/java/br/com/controllers/ProdutoControllerTest.java
package br.com.controllers;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import br.com.model.entities.classes.Produto;
import br.com.newObjects.model.entities.classes.NovoProduto;

@DataJpaTest
@AutoConfigureTestDatabase(replace = Replace.NONE)
public class ProdutoControllerTest {

    @Autowired
    private ProdutoController produtoController;

    @Test
    void testFind() throws Exception {
        var a = new Produto();
        for (int i = 0; i < 5; i++) {
            var b = produtoController.save(NovoProduto.novoProduto());
            if(i==3){
                a = b;
            }
        }
        System.out.println(produtoController.find(a));
    }

    @Test
    void testFindAll() throws Exception {
        for (int i = 0; i < 5; i++) {
            produtoController.save(NovoProduto.novoProduto());
        }
        System.out.println(produtoController.findAll());
    }

    @Test
    void testFindById() {

    }

    @Test
    void testRemove() {
        produtoController.remove(NovoProduto.novoProduto());
    }

    @Test
    void testSave() throws Exception {
        produtoController.save(NovoProduto.novoProduto());
    }

    @Test
    void testUpdate() {

    }
}

File: src/test/java/br/com/controllers/ProdutoControllerTest.kt
package br.com.controllers

import br.com.newObjects.model.NovoProduto.novoProduto
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class ProdutoControllerTest(@Autowired private val produtoController: ProdutoController) {
    @Test
    fun testFind() {
        var a = novoProduto()
        for (i in 0..4) {
            val b = produtoController.save(novoProduto())
            if (i == 3) {
                a = b
            }
        }
        println(a.id?.let { produtoController.findById(it) })
    }

    @Test
    fun testFindAll() {
        for (i in 0..4) {
            produtoController.save(novoProduto())
        }
        println(produtoController.findAll())
    }

    @Test
    fun testFindById() {
    }

    @Test
    fun testRemove() {
        produtoController.remove(novoProduto())
    }

    @Test
    fun testSave() {
        produtoController.save(novoProduto())
    }

    @Test
    fun testUpdate() {
    }
}
